### ISSUE TITLE
AbortSignals from Deno fetch do not work

### ISSUE THREAD

**Issue by bdashore3** [denoland/deno#26924]
Version: Deno 2.0.6

Hi there,
I'm trying to create a Hono server with a disconnect handler. When I cancel the request from my Postman client, the abort signal from the raw `Request` object never changes to true. This code works on both Hono and Bun, so I think this is an issue with Deno's fetch API. See example below (requires Hono)

```ts
import { Hono } from 'hono'

const app = new Hono()

async function stall(stallTime = 3000) {
  await new Promise(resolve => setTimeout(resolve, stallTime));
}

app.get('/hello', async (c) => {
  c.req.raw.signal.addEventListener("abort", () => {
    console.log("Aborted");
  })
  await stall(10000);
  return c.text("Hello from Hono!");
})

const port = 3000
console.log(`Server is running on http://localhost:${port}`)

Deno.serve(app.fetch)
```

**Comment by bartlomieju** [comment]
@bdashore3 are you sure you are on 2.0.6? It works fine for me and this was fixed in https://github.com/denoland/deno/pull/26781

**Comment by bdashore3** [comment]
Hi @bartlomieju!

Yes, I'm on Deno v2.0.6. Here's my console info. Should've mentioned that I'm on Windows. Also, that example does not work on my system.

```
deno 2.0.6 (stable, release, x86_64-pc-windows-msvc)
v8 12.9.202.13-rusty
typescript 5.6.2
```

**Comment by bdashore3** [comment]
However, this does work on the macOS version of Deno. I cannot test Linux, but it should be working there as well if it works on Mac.

```
deno 2.0.6 (stable, release, aarch64-apple-darwin)
v8 12.9.202.13-rusty
typescript 5.6.2
```

**Comment by Seally** [comment]
When I ran your example, the "Aborted" message was correctly printed to the console. I accessed the page on Firefox and closed the tab before the timeout. I'm on Windows 10.

Does the non-aborted response work as expected for you? 

It's a bit of a stab in the dark, but is there some security software on your Windows system that whitelists Node and Bun but not Deno (or something along those lines)? I remember someone from ages back had issues because their security software had sandboxed Deno but not Node.

**Comment by bdashore3** [comment]
@Seally I tried your suggestion and also reinstalled deno via winget instead. Now, the aborted print shows up.

However, there still seems to be some differences in behavior in this example between Windows and macOS. On Windows, the abort handler only triggers when the long-running task (await stall) is completed. However, on Mac, aborting the request immediately fires the event (which is what should happen).

I am not sure if my code is incorrect here. If it isn't, then there's something going on with how the event loop is prioritizing request cancellations on Windows vs other operating systems or there's something different happening with Windows request APIs.

To create a testable environment, I'd recommend using Postman or curl instead of a browser window.

**Comment by bdashore3** [comment]
Here's a more fine-grained snippet. I removed the stall function and used Deno's `delay` function

```ts
import { Hono, HonoRequest } from '@hono/hono'
import { logger } from "@hono/hono/logger"
import { delay } from "@std/async"

const app = new Hono()
app.use(logger())

async function checkDisconnect(req: HonoRequest) {
  while(!req.raw.signal.aborted) {
    await delay(1000);
  }

  console.log("Aborted");
  return;
}

app.get('/hello', async (c) => {

  // Simulate a long-running task
  // The cancel should immediately happen and finish the request
  // On Mac, this behavior is correct, but on Windows, the cancellation never fires
  await Promise.race([
    delay(10000),
    checkDisconnect(c.req),
  ]);

  return c.text("Hello from Hono!");
})

Deno.serve(app.fetch)
```

Basically, when cancelling the request from the client, the race will end and the function will finish. This behavior is correct on Mac, but not on Windows where the long-running task runs to its entirety.

**Comment by Seally** [comment]
I ran the latter example on Windows 10 (Deno v2.0.6, installed through chocolatey) and it gave me this output:

```
> deno run --allow-net main.ts
Listening on http://localhost:8000/
<-- GET /hello
Aborted
--> GET /hello 200 6s
<-- GET /hello
--> GET /hello 200 10s
Aborted
```

The first request I cancelled early while for the latter I let it run for the full 10 seconds. I tried running the code in Node and it didn't print the second `Aborted` message, which I assume to be the issue?

**Comment by bdashore3** [comment]
Fair enough. The main point I was trying to make there is that the request doesn't get cancelled until the delay is completed (which shouldn't happen). Thanks for looking into it! Hopefully the differing behaviors between OSes can be fixed.

**Comment by littledivy** [comment]
Why not use the signal event handler? This triggers even if the request is cancelled before delay:

```js
import { delay } from "jsr:@std/async";

function checkDisconnect(req) {
  const { promise, resolve } = Promise.withResolvers<void>();
  req.signal.addEventListener("abort", () => {
    resolve();
    console.log("Aborted");
  });

  return promise;
}

Deno.serve(async (req) => {
  await Promise.race([
    delay(10000),
    checkDisconnect(req)
  ]);

  return new Response("Ok");
});
```
